book_source/developers_guide/Logging.Rmd

```diff
@@ -0,0 +1,86 @@
+
+### Examples
+
+* The file [test.logger.R](../blob/master/utils/inst/tests/test.logger.R) provides descriptive examples
+* This query provides an current overview of [functions that use logging](https://github.com/PecanProject/pecan/search?q=logger&ref=cmdform)
+* logger functions (in order of increasing level):
+ * `logger.debug`
+ * `logger.info`
+ * `logger.warn`
+ * `logger.error`
+* the `logger.setLevel` function sets the level at which a message will be printed
+ * `logger.setLevel("DEBUG")` will print messages from all logger functions
+ * `logger.setLevel("ERROR")` will only print messages from `logger.error`
+ * `logger.setLevel("INFO")` and `logger.setLevel("WARN") shows messages from `logger.<level>` and higher functions, e.g. `logger.setLevel("WARN")` shows messages from `logger.warn` and `logger.error` 
+ * `logger.setLevel("OFF")` suppresses all logger messages
+* To print all messages to console, use `logger.setUseConsole(TRUE)`
+
+### Related Issues (requires Redmine developer account)
+
+* [#1071 How to handle errors?](https://ebi-forecast.igb.illinois.edu/redmine/issues/1071)
+* [#1222 Ignore warnings](https://ebi-forecast.igb.illinois.edu/redmine/issues/1222)
+  You can use @logger.setLevel("ERROR")@ to only show error messages. All the code that does not use logger will not be filtered.
+
+## Other R logging packages 
+
+* **This section is for reference - these functions should not be used in PEcAn, as they are redundant with the `logger.*` functions described above**
+
+R does provide a basic logging capability using stop, warning and message. These allow to print message (and stop execution in case of stop). However there is not an easy method to redirect the logging information to a file, or turn the logging information on and off. This is where one of the following packages comes into play. The packages themselves are very similar since they try to emulate log4j.
+
+Both of the following packages use a hierarchic loggers, meaning that if you change the level of displayed level of logging at one level all levels below it will update their logging.
+
+### logging
+
+The logging development is done at http://logging.r-forge.r-project.org/ and more information is located at http://cran.r-project.org/web/packages/logging/index.html . To install use the following command:
+<pre>
+install.packages("logging", repos="http://R-Forge.R-project.org")
+</pre>
+This has my preference pure based on documentation.
+
+### futile
+
+The second logging package is http://cran.r-project.org/web/packages/futile.logger/ and is eerily similar to logging (as a matter of fact logging is based on futile).
+
+## Example Usage
+
+To be able to use the loggers there needs to be some initialization done. Neither package allows to read it from a configuration file, so we might want to use the pecan.xml file to set it up. The setup will always be somewhat the same:
+
+```r
```

Comment:
We will eventually want this code to run, so these should be changed to the appropriate knitr R code chunk syntax:

    ```{r loggingexample1, echo = TRUE}
    library(logging)
    ...
    ```